sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -528,43 +530,52 @@ pi_result cuda_piPlatformsGet(pi_uint32 num_entries, pi_platform *platforms,
                               pi_uint32 *num_platforms) {
 
   try {
-    static constexpr pi_uint32 numPlatforms = 1;
+    static std::once_flag initFlag;
+    static pi_uint32 numPlatforms = 1;
+    static _pi_platform platformId;
 
-    if (num_platforms != nullptr) {
-      *num_platforms = numPlatforms;
+    if (num_entries == 0 and platforms != nullptr) {
```

Comment:
Surprised by the `and` - have never seen this before. What are the benefits (other than readability)?

Reply:
Do you mean the use of `and` rather than `&&` ?
Just readability AFAIK (see e.g. https://en.cppreference.com/w/cpp/language/operator_alternative)